Rakefile

```
@@ -1,43 +1,89 @@
+# frozen_string_literal: true
+
+require 'bundler'
+require 'puppet_litmus/rake_tasks' if Bundler.rubygems.find_name('puppet_litmus').any?
 require 'puppetlabs_spec_helper/rake_tasks'
-require 'puppet/version'
-require 'puppet/vendor/semantic/lib/semantic' unless Puppet.version.to_f < 3.6
-require 'puppet-lint/tasks/puppet-lint'
 require 'puppet-syntax/tasks/puppet-syntax'
-require 'metadata-json-lint/rake_task'
+require 'puppet_blacksmith/rake_tasks' if Bundler.rubygems.find_name('puppet-blacksmith').any?
+require 'github_changelog_generator/task' if Bundler.rubygems.find_name('github_changelog_generator').any?
+require 'puppet-strings/tasks' if Bundler.rubygems.find_name('puppet-strings').any?
 
-# These gems aren't always present, for instance on Travis with --without development
-begin
-  require 'puppet_blacksmith/rake_tasks'
-rescue LoadError # rubocop:disable Lint/HandleExceptions
+def changelog_user
+  return unless Rake.application.top_level_tasks.include? "changelog"
+  returnVal = nil || JSON.load(File.read('metadata.json'))['author']
+  raise "unable to find the changelog_user in .sync.yml, or the author in metadata.json" if returnVal.nil?
+  puts "GitHubChangelogGenerator user:#{returnVal}"
+  returnVal
 end
 
-if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.2')
-  require 'rubocop/rake_task'
+def changelog_project
+  return unless Rake.application.top_level_tasks.include? "changelog"
 
-  RuboCop::RakeTask.new
-end
+  returnVal = nil
+  returnVal ||= begin
```

Why is the first line needed here instead of the following?
```suggestion
  returnVal = begin
```

---

The Rakefile is generated from: https://github.com/puppetlabs/pdk-templates/blob/main/moduleroot/Rakefile.erb

So there are some PDK options that change these lines.